services/src/main/java/org/keycloak/protocol/oid4vc/issuance/signing/VCSigningServiceSpi.java

```diff
@@ -0,0 +1,32 @@
+package org.keycloak.protocol.oid4vc.issuance.signing;
+
+import org.keycloak.provider.Provider;
+import org.keycloak.provider.ProviderFactory;
+import org.keycloak.provider.Spi;
+
+/**
+ * @author <a href="https://github.com/wistefan">Stefan Wiedemann</a>
+ */
+public class VCSigningServiceSpi implements Spi {
+    private static final String NAME = "vcSigningService";
+
+    @Override
+    public boolean isInternal() {
+        return true;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public Class<? extends Provider> getProviderClass() {
+        return VerifiableCredentialsSigningService.class;
+    }
+
+    @Override
+    public Class<? extends ProviderFactory> getProviderFactoryClass() {
+        return VCSigningServiceProviderFactory.class;
```

Comment:
Is it possible to make sure that the class referenced from here is an interface rather than abstract class? It currently works with an abstract class, but AFAIK it is not recommended pattern and it is possible this can cause issues in the future (if Keycloak later does some refactoring in providers initialization part etc).

Hopefully just something minor like having interface `VCSigningServiceProviderFactory` and rename current abstract class to `AbstractVCSigningServiceProviderFactory` will do the job?

Reply:
fixed by [#7d5dea4](https://github.com/keycloak/keycloak/pull/25947/commits/7d5dea418e6cd92a3cb7b7577a545116a9e49ce8)